flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -266,6 +266,8 @@ public TaskExecutor(
 		this.currentRegistrationTimeoutId = null;
 
 		this.stackTraceSampleService = new StackTraceSampleService(rpcService.getScheduledExecutor());
+
+		taskExecutorServices.getShuffleEnvironment().setPartitionFailedOrFinishedListener(this::notifyFinalizedPartition);
```

Comment:
it is better to put this close with above `this.shuffleEnvironment = taskExecutorServices.getShuffleEnvironment();`